src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -7536,8 +7537,8 @@ final TDSReader startResponse(boolean isAdaptive) throws SQLServerException {
         // If command execution is subject to timeout then start timing until
         // the server returns the first response packet.
         if (queryTimeoutSeconds > 0) {
-            this.timeoutCommand = new TdsTimeoutCommand(queryTimeoutSeconds, this, null);
-            TimeoutPoller.getTimeoutPoller().addTimeoutCommand(this.timeoutCommand);
+            this.timeoutCommand = new TdsTimeoutCommand(queryTimeoutSeconds, this, tdsReader.getConnection());
```

Comment:
I'm wondering if passing connection is the right approach for calling `SQLServerTimeoutManager.startTimeoutCommand()` and would it be better to simply call `startTimeoutCommand()` with this instance as you're calling `releaseTimeoutCommand()`?

Reply:
I think that would be more ideal, but I'll have to do something different inside of the timeout manager if we want to still follow the whole recommended approach where the connection on close releases the timeout commands.

Specifically make some sort of map of connection-> timeout commands and call release with the connection everytime close is called.

Thoughts?

Reply:
@cheenamalhotra I made requested changes, please let me know of any other issues/concerns